Changes in order to get it to compile for Microsoft windows compilers

jimsch opened the pull request:
A set of changes to get it to compile with Visual Studio.  No Visual Studio solution files have been included, you will need to build your own.

jimsch committed:
Changes to build on windows

Some changes were needed to get a built working under Microsoft Windows
compilers/runtimes.

* Change a couple of header files
* Change the use of inline to be a windows keyword if using a windows
compiler.
* Create a definition for ssize_t in windows as this is a POSIX
constructu

jimsch committed:
Missed a file in the last commit

See previous comment.

hildjj commented:
Note: you should be able to generate visual studio files with cmake.  Something like:

```
cmake .. -G "Visual Studio 10 Win64"
```

jimsch commented:
Your are making an assumption that I am on a system that has cmake.  I am running on pure windows at the moment.  However adding files to a new project is trivial so I just did not worry about it.

jimsch commented:
This change does address Issue #26 for windows systems.

hildjj commented:
Download cmake Win32 binaries here: http://www.cmake.org/download/

we could also checkin the outputs I suppose.

jimsch committed:
Get cmake working on Windows

Changes to get the cmake build system do the right thing on windows
Start making changes to deal with the sizeof(int) != 8 problem

jimsch committed:
Missed file

Add cmake file

cabo commented:
I would like not to mix the 64-bit fixes (which are needed for all systems) with the fixes around lacking unistd.h etc.

jimsch committed:
Back Merge

Merge to current

jimsch committed:
Kill unused file

Delete #1 - emacs temp file

jimsch committed:
Fix merge problem

Fix merge problem

jimsch committed:
Change headers for MYLIB_EXPORT to be cleaner

Still should do a rename of this macro to be more correct

jimsch commented:
Created a new pull request with just the header file changes.

jimsch committed:
Merge remote-tracking branch 'refs/remotes/cabo/master' into windows

Conflicts:
	include/cn-cbor/cn-cbor.h
	src/cn-cbor.c
	src/cn-get.c

jimsch committed:
Change location of files

Does this help?

jimsch committed:
WHy does it not work?

jimsch committed:
Missed items during merge

Missed some changes during the merge step

jimsch committed:
Define macro for non-Windows platforms

jimsch committed:
Fix bounds checking on buffer
